docs/pmd_release_notes.md

```diff
@@ -0,0 +1,332 @@
+| UnusedNullCheckInEquals | Unused null check in equals | MAJOR | MAJOR | Active | Active |  |
+| UnusedPrivateField | Unused private field | MAJOR | MAJOR | DEPRECATED | Active | [S1068](https://rules.sonarsource.com/java/RSPEC-1068) |
+| UnusedPrivateMethod | Unused private method | MAJOR | MAJOR | DEPRECATED | Active | `java:UnusedPrivateMethod` |
+| UseArrayListInsteadOfVector | Use array list instead of vector | MAJOR | MAJOR | DEPRECATED | Active | [S1149](https://rules.sonarsource.com/java/RSPEC-1149) |
+| UseArraysAsList | Use arrays as list | MAJOR | MAJOR | Active | Active |  |
+| UseCollectionIsEmpty | Use collection is empty | MINOR | MAJOR | DEPRECATED | Active | [S1155](https://rules.sonarsource.com/java/RSPEC-1155) |
+| UseConcurrentHashMap | Use concurrent hash map | MAJOR | MAJOR | Active | Active |  |
+| UseCorrectExceptionLogging | Use correct exception logging | MAJOR | MAJOR | DEPRECATED | Active | [S1166](https://rules.sonarsource.com/java/RSPEC-1166) |
+| UseEqualsToCompareStrings | Use equals to compare strings | MAJOR | MAJOR | DEPRECATED | Active | `java:StringEqualityComparisonCheck`, [S1698](https://rules.sonarsource.com/java/RSPEC-1698) |
+| UseIndexOfChar | Use index of char | MAJOR | MAJOR | Active | Active |  |
+| UseLocaleWithCaseConversions | Use locale with case conversions | MAJOR | MAJOR | Active | Active |  |
+| UseNotifyAllInsteadOfNotify | Use notify all instead of notify | MAJOR | MAJOR | DEPRECATED | Active | [S2446](https://rules.sonarsource.com/java/RSPEC-2446) |
+| UseObjectForClearerAPI | Use object for clearer API | MINOR | MAJOR | DEPRECATED | Active | [S107](https://rules.sonarsource.com/java/RSPEC-107) |
+| UseProperClassLoader | Use proper class loader | CRITICAL | MAJOR | Active | Active |  |
+| UseStringBufferForStringAppends | Use string buffer for string appends | MAJOR | MAJOR | Active | Active |  |
+| UseStringBufferLength | Use string buffer length | MINOR | MAJOR | Active | Active |  |
+| UseUtilityClass | Use utility class | MAJOR | MAJOR | DEPRECATED | Active | [S1118](https://rules.sonarsource.com/java/RSPEC-1118) |
+| UseVarargs | Use varargs | MAJOR | MINOR | Active | Active |  |
+| UselessOperationOnImmutable | Useless operation on immutable | CRITICAL | MAJOR | Active | Active |  |
+| UselessOverridingMethod | Useless overriding method | MAJOR | MAJOR | DEPRECATED | Active | [S1185](https://rules.sonarsource.com/java/RSPEC-1185) |
+| UselessParentheses | Useless parentheses | INFO | MINOR | DEPRECATED | Active | `java:UselessParenthesesCheck` |
+| UselessQualifiedThis | Useless qualified this | MAJOR | MAJOR | Active | Active |  |
+| UselessStringValueOf | Useless string value of | MINOR | MAJOR | DEPRECATED | Active | [S1153](https://rules.sonarsource.com/java/RSPEC-1153) |
+
+## Skipped Deprecated Rules
+The following deprecated rules were skipped because they reference other rules (they are renamed):
```

Comment:
I am puzzled by what you mean by skipped. Unclear to me. Were they in the rule set before and not anymore? Were they deprecated, it doesn't show.

Reply:
These are rules in source pmd rules xml that have the following format:

```
<rule name="JUnit4TestShouldUseAfterAnnotation" deprecated="true" ref="UnitTestShouldUseAfterAnnotation" />
```

So with `deprecated="true"` and `ref="..."` and no other fields.